added phpunit 6

kokspflanze opened the pull request:
updated some tests with ::class
changed composer.json with phpunit to min 5.7.14 or 6.0.7

kokspflanze committed:
added phpunit 6
updated some tests with ::class
changed composer.json with phpunit to min 5.7.14 or 6.0.7

kokspflanze committed:
remove ignore-platform-reqs in travis

froschdesign commented on test/Test/EventListenerIntrospectionTraitTest.php:
```diff
@@ -193,6 +194,13 @@ public function testAssertListenerAtPriorityFailsWhenListenerIsNotFound()
                     $arguments['event'],
                     $arguments['priority']
                 ), $e->getMessage(), sprintf('Assertion failure message was unexpected: %s', $e->getMessage()));
+            // for PHPUnit 5.7
```
Please no dependencies to a specific version.

kokspflanze replied:
thats the fallback for PHPUnit5, thats not just phpunit5.7

that exception class doesnt exist in phpunit 6, so there must be a fallback, other way would be a `class_alias` or a check if the class exists.

michalbundyra replied:
Please have a look how @weierophinney did it in another repo:
https://github.com/zendframework/zend-component-installer/commit/654983e8f01ddb636c1886fa1880db854d9cc0bd
I'd suggest the same way here.

kokspflanze replied:
ohh i missed that, is that workflow is expected, than i will change that sure =)

michalbundyra commented on src/Test/EventListenerIntrospectionTrait.php:
```diff
@@ -9,7 +9,7 @@
 namespace Zend\EventManager\Test;
 
-use PHPUnit_Framework_Assert as Assert;
+use PHPUnit\Framework\Assert as Assert;
```
Alias `Assert` here is redundant.

michalbundyra commented:
@kokspflanze We need also update travis configuration, to install PHPUnit 5.7 on PHP5.6, please see changes for example here:
https://github.com/zendframework/zend-component-installer/blob/master/.travis.yml#L16
and `install` section there.

michalbundyra commented on composer.json:
```diff
@@ -32,7 +32,7 @@
     },
     "require-dev": {
-        "phpunit/PHPUnit": "^5.6",
+        "phpunit/PHPUnit": "^5.7.14 || ^6.0.7",
```
I would suggest also swap order here, please see:
https://github.com/zendframework/zend-expressive-skeleton/commit/39ef6894a52c54bc030bcca51f051d41a3bcbe61

kokspflanze committed:
updated phpunit 5.7 fallback
Alias Assert is redundant
update travis configuration
swap order of phpunit version

kokspflanze committed:
added check for phpunit 5.7 fallback
updated doc-block

kokspflanze committed:
cs part fixed

kokspflanze commented:
done, i just added in the autoload file a `class_exist` check, to remove the warning

michalbundyra reviewed (approved):
LGTM 👍

marc-mabe commented on src/Test/EventListenerIntrospectionTrait.php:
```diff
@@ -9,7 +9,7 @@
 namespace Zend\EventManager\Test;
 
-use PHPUnit_Framework_Assert as Assert;
+use PHPUnit\Framework\Assert;
```
Why not using `$this->assertTrue()` instead of `Assert::assertTrue()`

kokspflanze replied:
why?

in that case we are in a trait class and the class that this trait import can not have that method.

marc-mabe replied:
Makes sense

froschdesign replied:
> Makes sense

Sure? If we look at the comments of the class:

> Trait providing utility methods and assertions for use in PHPUnit test cases.

😉

kokspflanze replied:
before the `Assert` class it was with the `TestCase` class

https://github.com/zendframework/zend-eventmanager/pull/19/commits/be75ec1c3ddafaeb56f99c654a8968ec8463c041

from my site i see the `Assets` class as better solution, with `$this` it could be possible that it break smth in other tests if someone not use the `TestCase` as parent-class. also we have a autocomplete and no warning that the method not exist.

marc-mabe replied:
@froschdesign In a trait it's only possible to define an abstract method to define that a specific method needs to be available but that does not work with `__call()` so it gets marked red in *good* IDEs. Using the direct assert seems to be more clear and the trait itself can be tested much better.

froschdesign replied:
I fine with it. No extra discussion needed. (Look at the smiley!)

froschdesign commented:
@kokspflanze 

The topic of this PR is "added phpunit 6", but you provide an all-in-one-commit:

- namespace imports
- class names with `::class`
- `expectException` and `expectExceptionMessage`
- fallback

This may be necessary, but not in **one** commit! It's fine to add all this changes to one PR, but please create separate commits for each problem. We work with a version control system and these all-in-one-commits breaks this system. Try a cherry-pick or try to read the history.

**Each individual change should be one commit.**

(Added benefit: a review becomes easier and faster!)

kokspflanze commented:
@froschdesign no problem to split it in more commits, but that week is no much space, hope its okay if i do that next week.

didnt know that this should be in more commits, it was for me one little task.

froschdesign commented:
@kokspflanze 

>  …but that week is no much space, hope its okay if i do that next week.

I can live with _this_ PR. 

@marc-mabe What are you saying?

> didnt know that this should be in more commits, it was for me one little task.

Which mixes different problems!

---

*Each individual change should be one commit.*

That's not a specific feature or requirement of Zend Framework, GitHub or me. That's the principle of a version control system.

weierophinney commented:
Thanks, @kokspflanze
